cmd/skopeo/list_tags_test.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+// Tests the kinds of inputs allowed and expected to the command
+func TestDockerRepositoryReferenceParser(t *testing.T) {
+	for _, test := range [][]string{
+		{"docker://myhost.com:1000/nginx", "myhost.com:1000/nginx"},    //no tag
+		{"docker://myhost.com/nginx", "myhost.com/nginx"},              //no port or tag
+		{"docker://somehost.com", "docker.io/library/somehost.com"},    // Valid default expansion
+		{"docker://nginx", "docker.io/library/nginx"},                  // Valid default expansion
+		{"docker://myhost.com:1000/nginx:foobar:foobar", ""},           // Invalid repository ref
+		{"docker://somehost.com:5000/", ""},                            // no repo
+		{"docker://myhost.com:1000/nginx:latest", ""},                  //tag not allowed
+		{"docker://myhost.com:1000/nginx@sha256:abcdef1234567890", ""}, //digest not allowed
+	} {
+
+		ref, err := parseDockerRepositoryReference(test[0])
+
+		if test[1] == "" {
+			assert.Error(t, err, "Expected error in parsing but no error raised for %v", test[0])
+		} else {
+			if assert.NoError(t, err, "Could not parse, got error on %v", test[0]) {
+				assert.Equal(t, test[1], ref.DockerReference().Name(), "Mismatched parse result for input %v", test[0])
+			}
+		}
+	}
+}
+
+func TestDockerRepositoryReferenceParserDrift(t *testing.T) {
+	for _, test := range [][]string{
+		{"docker://myhost.com:1000/nginx", "myhost.com:1000/nginx"},    //no tag
+		{"docker://myhost.com/nginx", "myhost.com/nginx"},              //no port or tag
+		{"docker://somehost.com", "docker.io/library/somehost.com"},    // Valid default expansion
+		{"docker://nginx", "docker.io/library/nginx"},                  // Valid default expansion
+	} {
+
+		ref, err := parseDockerRepositoryReference(test[0])
+		ref2, err2 := alltransports.ParseImageName(test[0])
+
+		if assert.NoError(t, err, "Could not parse, got error on %v", test[0]) && assert.NoError(t, err2, "Could not parse with regular parser, got error on %v", test[0]) {
+			assert.Equal(t, ref.DockerReference().String(), ref2.DockerReference().String(), "Different parsing output for input %v. Repo parse = %v, regular parser = %v", test[0], ref, ref2)
+		}
+	}
+}
+
+func TestUnsupportedRepositoryReferenceParser(t *testing.T) {
```

Comment:
(Non-blocking: This can just as well be a part of `TestDockerRepositoryReferenceParser`.)